lib/realtime/rtr.ex

```diff
@@ -3,14 +3,93 @@ defmodule Realtime.RTR do
   for parsing data out of RTR's json feeds
   """
 
-  # TODO: create structs and alias them as necessary
+  alias Realtime.Data.TripUpdate
+  alias Realtime.Data.VehiclePosition
+  alias Realtime.PollingServer
 
-  def parse_data(filedata) do
+  # Parse VehiclePositions from the enhanced json file. Special cased for empty files.
+  @spec parse_vehicle_positions(String.t()) ::
+          PollingServer.realtime_info(VehiclePosition.t()) | nil
+  def parse_vehicle_positions(""), do: nil
+
+  def parse_vehicle_positions(filedata) do
+    json = Jason.decode!(filedata)
+
+    %{
+      timestamp: json["header"]["timestamp"],
+      entities:
+        json["entity"]
+        |> Enum.map(&parse_vehicle_position/1)
+        |> Enum.filter(fn vp -> vp.route_id != nil end)
```

Comment:
note: Ah, here is where `route_id` cannot be `nil`. So that can be reflected in the spec 👍